app/models/pusher.rb

```diff
@@ -20,6 +20,11 @@ def authorize
       notify("You do not have permission to push to this gem. Ask an owner to add you with: gem owner #{rubygem.name} --add #{user.email}", 403)
   end
 
+  def verify_mfa_requirement
+    rubygem.mfa_requirement_satisfied_for?(user) ||
+      notify("Rubygem requires owners to enable MFA. You must enable MFA before pushing new version.", 403)
```

Comment:
Why are we not checking `spec.metadata["rubygems_mfa_required"]` here? To me it seems more intuitive that MFA requirement is enforced from the same version we set this.